### DISCUSSION TITLE
Support filtering of alerting rules by label in `/api/v1/rules`

### DISCUSSION THREAD

**Discussion by K-Phoen** [grafana/grafana#49129] [Alerting]
It is not unusual for companies to rely on Grafana's API to extract information on the state of the alerting across teams. For example, to display in developer portals (such as [Backstage](https://backstage.io/)) or to get some key engineering metrics.

Usual needs include:
* list of alerting rules defined for service/team/system/domain X
* list of SLO-related alerting rules
* ...

With the legacy alerting, labels were a good way to get these alerting rules across dashboards and folders using the `dashboardTag` parameter on the [`GET /api/alerts/` HTTP endpoint](https://grafana.com/docs/grafana/latest/http_api/alerting/#get-alerts).

The [`/api/ruler/grafana/api/v1/rules`](https://editor.swagger.io/?url=https://raw.githubusercontent.com/grafana/grafana/main/pkg/services/ngalert/api/tooling/post.json#operations-tag-ruler) endpoint exposed by the unified alerting API does not provide similar capabilities and instead requires users to fetch all the rules and filter them after the fact.

If this a pull request implementing this would be considered, I can open one (based on this [initial implementation](https://github.com/K-Phoen/grafana/commit/e539b988a9d82ce707458662575be15f5e275830))

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
